ui/pages/confirmations/hooks/useCurrentConfirmation.ts

```diff
@@ -55,25 +60,29 @@ const useCurrentConfirmation = () => {
     selectUnapprovedMessage(state, confirmationId),
   );
 
-  const isCorrectTransactionType =
-    isRedesignedConfirmationsDeveloperSettingEnabled &&
-    REDESIGN_TRANSACTION_TYPES.includes(
-      transactionMetadata?.type as TransactionType,
-    );
+  const isCorrectTransactionType = REDESIGN_TRANSACTION_TYPES.includes(
+    transactionMetadata?.type as TransactionType,
+  );
 
   const isCorrectApprovalType = REDESIGN_APPROVAL_TYPES.includes(
     pendingApproval?.type as ApprovalType,
   );
 
-  const isSIWE =
-    pendingApproval?.type === TransactionType.personalSign &&
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    (signatureMessage?.msgParams as any)?.siwe?.isSIWEMessage;
+  const shouldUseRedesignForSignatures =
+    isRedesignedSignaturesUserSettingEnabled && isCorrectApprovalType;
+
+  const shouldUseRedesignForTransactions =
+    isRedesignedTransactionsUserSettingEnabled && isCorrectTransactionType;
 
+  // If the developer toggle or the build time environment variable are enabled,
+  // all the signatures and transactions in development are shown. If the user
+  // facing feature toggles for signature or transactions are enabled, we show
+  // only confirmations that shipped (contained in `REDESIGN_APPROVAL_TYPES` and
+  // `REDESIGN_TRANSACTION_TYPES` respectively).
   const shouldUseRedesign =
-    isRedesignedConfirmationsUserSettingEnabled &&
-    (isCorrectApprovalType || isCorrectTransactionType) &&
-    !isSIWE;
+    shouldUseRedesignForSignatures ||
+    shouldUseRedesignForTransactions ||
+    isRedesignedConfirmationsDeveloperSettingEnabled;
```

Comment:
Do we not want to always validate the approval type and transaction type?

Otherwise if using the ENV or developer setting, it will try use the redesigned confirmation for types not in `REDESIGN_TRANSACTION_TYPES` and `REDESIGN_APPROVAL_TYPES`?

Reply:
Good catch Matthew. Updated.